rundeckapp/grails-spa/packages/ui-trellis/src/library/components/storage/tests/KeyStorageEdit.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { mount } from "@vue/test-utils";
+import KeyStorageEdit from "../KeyStorageEdit.vue";
+
+const mountKeyStorageEdit = async (props = {}) => {
+  return mount(KeyStorageEdit, {
+    props: {
+      storageFilter: "",
+      uploadSetting: {
+        modifyMode: false,
+        keyType: "privateKey",
+        inputPath: "",
+        inputType: "text",
+        fileName: null,
+        file: null,
+        fileContent: "",
+        textArea: "some-text",
+        password: "",
+        status: "new",
+        errorMsg: null,
+        dontOverwrite: false,
+      },
+      project: "test-project",
+      ...props,
+    },
+  });
+};
+describe("KeyStorageEdit", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("emits cancelEditing when cancel button is clicked", async () => {
+    const wrapper = await mountKeyStorageEdit();
+    const cancelButton = wrapper.find('[data-testid="cancel-btn"]');
+    await cancelButton.trigger("click");
+    expect(wrapper.emitted("cancelEditing")).toHaveLength(1);
+    expect(Object.keys(wrapper.emitted())).toContain("cancelEditing");
+  });
+
+  it("emits finishEditing with correct data when Save button is clicked", async () => {
+    const wrapper = await mountKeyStorageEdit();
+    const result = {
+      success: true,
+      keyDetails: {
+        name: "exampleKey",
+        path: "/keys/exampleKey",
+        keyType: "privateKey",
+      },
+    };
+    jest.spyOn(wrapper.vm as any, "handleUploadKey").mockImplementation(() => {
+      wrapper.vm.$emit("finishEditing", result);
+    });
+    const saveButton = wrapper.find('[data-testid="save-btn"]');
+    await wrapper.vm.$nextTick();
+    expect(wrapper.emitted().finishEditing).toBeFalsy();
+    expect(saveButton.isVisible()).toBe(true);
```

Comment:
Save button is always visible, can instead check that isn't disabled Or just remove this expect since there's a test focused on the disabled aspect